### PR TITLE
Update to PHP 8.0 and PHP 8-compatible types.

### PR DESCRIPTION
**Pull request by Crell**

### CODE REVIEW

**Review by ashnazg — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0"
+        "php": ">=8.0.0"
```
Comment (Nyholm) [inline composer.json:13]: Interesting. This would allow to tag 1.1.. =)

Just saying.. I know you have a plan.

**Review by Nyholm — APPROVED** [review]

File: src/CacheItemInterface.php [hunk 2]
```diff
@@ -72,12 +72,12 @@ public function isHit();
...
      *   The invoked object.
      */
-    public function set($value);
+    public function set(mixed $value);
 
     /**
      * Sets the expiration time for this cache item.
      *
-     * @param \DateTimeInterface|null $expiration
+     * @param ?\DateTimeInterface $expiration
```
Comment (samdark) [inline src/CacheItemInterface.php:80]: Should stay as is: https://docs.phpdoc.org/3.0/guide/references/phpdoc/types.html

File: src/CacheException.php [hunk 3]
```diff
@@ -5,6 +5,6 @@
 /**
  * Exception interface for all exceptions thrown by an Implementing Library.
  */
-interface CacheException
+interface CacheException extends \Throwable
```
Comment (PrinsFrank) [inline src/CacheException.php:8]: Looks like the same change was missed for the InvalidArgumentException?

Comment (Nyholm) [reply src/CacheException.php:8]: The InvalidArgumentException implements CacheException, right?

Comment (PrinsFrank) [reply src/CacheException.php:8]: Sorry, nvm!

**Review by PrinsFrank — APPROVED** [review]

File: src/CacheException.php (hunk 3 above)
Comment (PrinsFrank) [reply src/CacheException.php:8]: I don't know how to close this one, approving doesn't help apparently.

File: src/CacheItemInterface.php (hunk 2 above)
Comment (Crell) [reply src/CacheItemInterface.php:80]: cf https://github.com/php-fig/cache/pull/24#discussion_r543536544

**Pull request merged**